splitio/sync/split.py

```diff
@@ -318,3 +354,168 @@ def synchronize_splits(self, till=None):  # pylint:disable=unused-argument
+        :rtype: Dict
+        """
+        parsed = self._sanitize_json_elements(parsed)
+        parsed['splits'] = self._sanitize_split_elements(parsed['splits'])
+
+        return parsed
+
+    def _sanitize_json_elements(self, parsed):
+        """
+        Sanitize all json elements.
+
+        :param parsed: splits, till and since elements dict
+        :type parsed: Dict
+
+        :return: sanitized structure dict
+        :rtype: Dict
+        """
+        if 'splits' not in parsed:
+            parsed['splits'] = []
+        if 'till' not in parsed or parsed['till'] is None or parsed['till'] < -1:
+            parsed['till'] = -1
+        if 'since' not in parsed or parsed['since'] is None or parsed['since'] < -1 or parsed['since'] > parsed['till']:
+            parsed['since'] = parsed['till']
+
+        return parsed
+
+    def _sanitize_split_elements(self, parsed_splits):
+        """
+        Sanitize all splits elements.
+
+        :param parsed_splits: splits array
+        :type parsed_splits: [Dict]
+
+        :return: sanitized structure dict
+        :rtype: [Dict]
+        """
+        sanitized_splits = []
+        for split in parsed_splits:
+            if 'name' not in split or split['name'].strip() == '':
+                _LOGGER.warning("A split in json file does not have (Name) or property is empty, skipping.")
+                continue
+            for element in [('trafficTypeName', 'user', None, None, None, None),
+                            ('trafficAllocation', 100, 0, 100,  None, None),
+                            ('trafficAllocationSeed', get_current_epoch_time_ms(), None, None, None, [0]),
+                            ('seed', get_current_epoch_time_ms(), None, None, None, [0]),
+                            ('status', splits.Status.ACTIVE.value, None, None, [e.value for e in splits.Status], None),
+                            ('killed', False, None, None, None, None),
+                            ('defaultTreatment', 'on', None, None, None, ['', ' ']),
+                            ('changeNumber', 0, 0, None, None, None),
+                            ('algo', 2, 2, 2, None, None)]:
+                split = util._sanitize_object_element(split, 'split', element[0], element[1], lower_value=element[2], upper_value=element[3], in_list=element[4], not_in_list=element[5])
+            split = self._santizie_condition(split)
+            sanitized_splits.append(split)
+        return sanitized_splits
+
+    def _santizie_condition(self, split):
+        """
+        Sanitize split and ensure a condition type ROLLOUT and matcher exist with ALL_KEYS elements.
+
+        :param split: split dict object
+        :type split: Dict
+
+        :return: sanitized split
+        :rtype: Dict
+        """
+        found_all_keys_matcher = False
+        if 'conditions' not in split or split['conditions'] is None:
+            split['conditions'] = []
+        for condition in split['conditions']:
```

Comment:
you need to check if the last condition matches with ALL_KEYS not if it has one.

Reply:
fixed